### PR TITLE
error handling - send message from graphql

### PR DESCRIPTION
**Pull request by kazukageorge**
Before:
```
(.venv) george@Georges-MacBook-Pro sparsezoo % python3 '/Users/george/neuralmagic/sparsezoo/scratch/graohql.py'
Traceback (most recent call last):
  File "/Users/george/neuralmagic/sparsezoo/scratch/graohql.py", line 15, in <module>
    m = gql.fetch(operation_body=operation_body, fields=fields, arguments=arguments)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 46, in fetch
    response_objects = self.make_request(
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 82, in make_request
    return response_json["data"][query.operation_body]
TypeError: 'NoneType' object is not subscriptable
```

After:
```
Traceback (most recent call last):
  File "/Users/george/neuralmagic/sparsezoo/scratch/graohql.py", line 15, in <module>
    m = gql.fetch(operation_body=operation_body, fields=fields, arguments=arguments)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 47, in fetch
    response_objects = self.make_request(
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/exceptions.py", line 33, in inner_function
    return fn(*args, **kwargs)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 81, in make_request
    validate_graphql_response(response=response, query_body=query.query_body)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/exceptions.py", line 46, in validate_graphql_response
    raise InvalidQueryException(f"{response_json['errors']}\n{query_body}")
sparsezoo.api.exceptions.InvalidQueryException: [{'message': "Cannot query field 'blah' on type 'Query'.", 'locations': [{'line': 3, 'column': 9}]}]

    {
        blah (architecture: "bert",)
            {
               benchmarkResults { batchSize deviceInfo numCores recordedUnits recordedValue } 
            }
    }

```

### CODE REVIEW

File: src/sparsezoo/api/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
```
Comment (derekk-nm) [inline src/sparsezoo/api/exceptions.py:1]: should this be 2023, not 2021?

File: src/sparsezoo/api/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        except HTTPError as _err:
+            logger.error(f"request exception: {_err}")
+
+        except InvalidQueryValueError as _err:
+            logger.error(f"InvalidQueryValueError: {_err}")
+
+    return inner_function
+
+
+def validate_graphql_response(response: Response, query_body: str) -> None:
```
Comment (derekk-nm) [inline src/sparsezoo/api/exceptions.py:50]: why wrap this code in a function?  you're only calling it once in the make_request method.  It seems that you can just use this function's lines in the make_request method and import InvalidQueryValueError there.

File: src/sparsezoo/api/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+logger = logging.getLogger(__name__)
+
+
+class InvalidQueryValueError(Exception):
+    def __init__(self, error_message: Optional[str] = None, *args, **kwargs):
+        super().__init__(
+            error_message,
+            *args,
+            **kwargs,
+        )
```
Comment (tdg5) [inline src/sparsezoo/api/exceptions.py:32]: Is this really needed? I'd think based on the content of this method that we wouldn't need to define an `__init__` method here at all

File: src/sparsezoo/api/exceptions.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+
+def graphqlapi_exception_handler(fn: Callable) -> Callable:
+    @wraps(fn)
+    def inner_function(*args, **kwargs):
+        try:
+            return fn(*args, **kwargs)
+
+        except HTTPError as _err:
+            logger.error(f"request exception: {_err}")
```
Comment (tdg5) [inline src/sparsezoo/api/exceptions.py:42]: I would think this should still `raise`? You might also consider `logger.exception(_err)`

File: src/sparsezoo/api/exceptions.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+        except InvalidQueryValueError as _err:
+            logger.error(f"InvalidQueryValueError: {_err}")
```
Comment (tdg5) [inline src/sparsezoo/api/exceptions.py:45]: I would think this should still raise? You might also consider logger.exception(_err)

File: src/sparsezoo/api/exceptions.py (hunk 3 above)
Comment (kazukageorge) [reply src/sparsezoo/api/exceptions.py:32]: without init the error will be 
```
(.venv) george@Georges-MacBook-Pro sparsezoo % python3 '/Users/george/neuralmagic/sparsezoo/scratch/graohql.py'
Traceback (most recent call last):
  File "/Users/george/neuralmagic/sparsezoo/scratch/graohql.py", line 15, in <module>
    m = gql.fetch(operation_body=operation_body, fields=fields, arguments=arguments)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 47, in fetch
    response_objects = self.make_request(
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/exceptions.py", line 40, in inner_function
    return fn(*args, **kwargs)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/graphql.py", line 81, in make_request
    validate_graphql_response(response=response, query_body=query.query_body)
  File "/Users/george/neuralmagic/sparsezoo/src/sparsezoo/api/exceptions.py", line 56, in validate_graphql_response
    raise InvalidQueryValueError(
TypeError: InvalidQueryValueError() takes no keyword arguments
(.venv) george@Georges-MacBook-Pro sparsezoo % 
```

Comment (kazukageorge) [reply src/sparsezoo/api/exceptions.py:32]: ah i got it

File: src/sparsezoo/api/exceptions.py (hunk 2 above)
Comment (kazukageorge) [reply src/sparsezoo/api/exceptions.py:41]: One reason is to decouple the logic, making it easier to test

File: tests/sparsezoo/api/test_exception.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        ),
+        (
+            False,
+            {
+                "data": "foo",
+            },
+        ),
+    ],
+)
+def test_whatever(should_raise_invalid_query_exception: bool, response: Dict):
```
Comment (tdg5) [inline tests/sparsezoo/api/test_exception.py:54]: May I request a better test name 🙈

**Review by tdg5 — APPROVED** [review]

**Review by dfneuralmagic — APPROVED** [review]
LGTM

**Pull request merged**